### DISCUSSION TITLE
Can I generate latex code from R and then render it in the same document?

### DISCUSSION THREAD

**Discussion by mattysimonson** [quarto-dev/quarto-cli#7539] [Q&A]
### Description

Suppose I have r code that generates a table in latex such as 

```{r}
library(xtable)
data(tli)
xtable(tli[1:10, ])
```
How can I get this table (rather than the code) to show up in my Quarto document? Do I have to save it as a separate .tex file and then load it back in?

**Comment by mcanouil** [comment]
The issue is that `xtable` does not actually returns the code but prints it.
This is pretty extensively "documented" online, see <https://duckduckgo.com/?q=xtable+rmarkdown>.

Also documented online, you need to tell `knitr` the print output needs to be read as is.

````qmd
---
format: pdf
---

```{r}
#| output: asis
library(xtable)
data(tli)
xtable(tli[1:10, ])
```
````

**Reply by cderv** [reply]
Additionally, A trick to set the printing method for **xtable** for a whole document without needing the options in each chunk

````markdown
```{r}
#| include: false
library(knitr)
.S3method("knit_print", "xtable", function(x, ...) {
  knitr::raw_latex(print(x, print.results = FALSE))
})
```
````

This can be adapted to account for different output format. 

**xtable** is among the few package that do not have some nice `knit_print` method to account for **knitr**.  But its `print()` method as a few options that can be tweaked to customize output (https://github.com/cran/xtable/blob/master/R/print.xtable.R)


Hope it helps
